assets/js/atomic/components/product/button/index.js

```diff
@@ -106,6 +106,18 @@ const ProductButton = ( { product, className } ) => {
 		}
 		return productCartDetails.text;
 	};
+
+	// This is a hack to trigger cart updates till we migrate to block based card
+	// that relies on the store, see
+	// https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1247
+	useEffect( () => {
+		// eslint-disable-next-line no-undef
+		const event = new Event( 'wc_fragment_refresh', {
+			bubbles: true,
+			cancelable: true,
+		} );
+		document.body.dispatchEvent( event );
+	}, [ addedToCart ] );
```

Comment:
I think replacing this line should fix the issue described above:
```suggestion
	}, [ cartQuantity ] );
```

Reply:
This seems to fix it